exec/plugin-commands-script-runners/src/exec.ts

```diff
@@ -226,6 +227,10 @@ export async function handler (
             return
           }
 
+          if (err.originalMessage === `spawn ${params[0]} ENOENT`) {
+            err.hint = buildCommandNotFoundHint(params[0], (await readProjectManifestOnly(opts.dir)).scripts)
+          }
```

Comment:
🤔 I couldn't figure out how to get testing for this to work. Is there some good way to get the test output?

Example test that fails:

```ts

test.only('pnpm exec command not found', async () => {
  prepare({
    scripts: {
      build: "echo hello"
    }
  })

  await expect(exec.handler({
    ...DEFAULT_OPTS,
    dir: process.cwd(),
    reportSummary: true,
    selectedProjectsGraph: {},
  }, ['buil'])).rejects
  
  // TODO: the error thrown doesn't seem to have code or hint?
  .toEqual(expect.objectContaining({
    code: 'ERR_PNPM_NO_SCRIPT',
    hint: 'Command "buil" not found. Did you mean "pnpm run build"',
  }))
});
```

```diff

    - ObjectContaining {
    -   "code": "ERR_PNPM_NO_SCRIPT",
    -   "hint": "Command \"buil\" not found. Did you mean \"pnpm run build\"",
    - }
    + [Error: "pnpm recursive exec" failed in 1 packages]
```

Reply:
maybe try a `try/catch` and check the `hint` property of the catched error.

Reply:
```plaintext
console.log
    got error.hint undefined

      at Object.<anonymous> (test/exec.e2e.ts:428:13)

  console.log
    got error RecursiveFailError: "pnpm recursive exec" failed in 1 packages
        at throwOnCommandFail (/Users/josh/repos/pnpm/cli/cli-utils/src/recursiveSummary.ts:31:11)
        at Object.handler (/Users/josh/repos/pnpm/exec/plugin-commands-script-runners/src/exec.ts:253:21)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at Object.<anonymous> (/Users/josh/repos/pnpm/exec/plugin-commands-script-runners/test/exec.e2e.ts:421:5) {
      code: 'ERR_PNPM_RECURSIVE_FAIL',
      hint: undefined,
      attempts: undefined,
      failures: [
        {
          status: 'failure',
          duration: 13.299,
          error: [Error],
          message: 'Command failed with ENOENT: buil\nspawn buil ENOENT',
          prefix: '/Users/josh/repos/pnpm_tmp/c764b3ed7661091f60f194fa52ecd314/2/project'
        }
      ],
      passes: 0
    }
```

😞

Reply:
I guess what I'm looking for is: any end-to-end test that runs `pnpm exec`, _not_ `pnpm recursive exec`? These ones all seem to me are running the recursive version, and thus not hitting the right code path? Apologies for not being very familiar with how that process works.